frontend/lib/src/components/core/Maybe/Maybe.test.tsx

```diff
@@ -28,65 +31,82 @@ interface OuterProps {
-    afterEach(() => {
-      jest.restoreAllMocks()
-    })
+  beforeEach(() => {
+    innerRenderCount = 0
+  })
 
-    it("should invoke the render method when the props of an enclosing element update", () => {
+  describe("when enable is true", () => {
+    it("should render when the props of an enclosing element update", () => {
       const { rerender } = render(<Outer name={"old again"} enable={true} />)
 
-      const spyShouldComponentUpdate = jest.spyOn(
-        Maybe.prototype,
-        "shouldComponentUpdate"
-      )
-      const spyRender = jest.spyOn(Maybe.prototype, "render")
+      expect(innerRenderCount).toBe(1)
+      expect(screen.getByText("old again")).toBeVisible()
 
       rerender(<Outer name={"new name"} enable={true} />)
 
-      expect(spyShouldComponentUpdate).toHaveBeenCalled()
-      expect(spyRender).toHaveBeenCalled()
+      expect(innerRenderCount).toBe(2)
+      expect(screen.getByText("new name")).toBeVisible()
     })
 
-    it("should call render() when a Maybe is first disabled", () => {
-      const { rerender } = render(<Outer name={"old again"} enable={true} />)
+    it("should update when a Maybe is first disabled", () => {
+      const { rerender } = render(<Outer name={"old again"} enable={false} />)
 
-      const spyShouldComponentUpdate = jest.spyOn(
-        Maybe.prototype,
-        "shouldComponentUpdate"
-      )
-      const spyRender = jest.spyOn(Maybe.prototype, "render")
+      expect(innerRenderCount).toBe(1)
+      expect(screen.getByText("old again")).toBeVisible()
 
-      rerender(<Outer name={"new name"} enable={false} />)
+      rerender(<Outer name={"new name"} enable={true} />)
 
-      expect(spyShouldComponentUpdate).toHaveBeenCalled()
-      expect(spyRender).toHaveBeenCalled()
+      expect(innerRenderCount).toBe(2)
+      // Because enable changes between renders, the inner component should be
+      // rerendered.
+      expect(screen.queryByText("new name")).toBeVisible()
     })
   })
 
   describe("when enable is false", () => {
-    it("should not invoke the render method when the props of an enclosing element update", () => {
+    it("should not render children when disabled", () => {
       const { rerender } = render(<Outer name={"old again"} enable={false} />)
 
-      const spyShouldComponentUpdate = jest.spyOn(
-        Maybe.prototype,
-        "shouldComponentUpdate"
-      )
-      const spyRender = jest.spyOn(Maybe.prototype, "render")
+      expect(innerRenderCount).toBe(1)
+      expect(screen.queryByText("old again")).toBeVisible()
 
       rerender(<Outer name={"new name"} enable={false} />)
 
-      expect(spyShouldComponentUpdate).toHaveBeenCalled()
-      expect(spyRender).toHaveBeenCalledTimes(0)
+      expect(innerRenderCount).toBe(1)
+      expect(screen.queryByText("old again")).toBeVisible()
+      expect(screen.queryByText("new name")).not.toBeInTheDocument()
+    })
+
+    it('does not re-render when "enable" is false', () => {
```

Comment:
the test body looks like its the same test as in "should not render children when disabled", except that `rerender(<Outer name={"old again"} enable={false} />)` is called more than once. In this case I am not 100% why the tests are split into two. Is this intended or do I miss something?

Reply:
That's right, these are pretty much the same. I have condensed them into 1 test to clean it up.